### DISCUSSION TITLE
Running desktop

### DISCUSSION THREAD

**Discussion by Raviu56** [devmfc/debian-on-amlogic#2] [Q&A]
Thank you very much for your work here.
Only last version 6.1.25 could boot on my s905x4 box.
As a text mode or headless server it works very well.
I've tried to run desktop-minimal-install.sh script on both ubuntu and debian version. and for both versions the desktop could not be started only blinking curser on black screen.
Stopping gdm service and running startx command, I get the following Xorg.0.log
```
[  4370.961] (WW) Falling back to old probe method for modesetting
[  4370.979] (WW) Falling back to old probe method for modesetting
[  4370.979] (EE) No devices detected.
[  4370.979] (==) Matched modesetting as autoconfigured driver 0
[  4370.979] (==) Matched fbdev as autoconfigured driver 1
[  4370.979] (==) Assigned the driver to the xf86ConfigLayout
[  4370.979] (II) LoadModule: "modesetting"
[  4370.979] (II) Loading /usr/lib/xorg/modules/drivers/modesetting_drv.so
[  4370.979] (II) Module modesetting: vendor="X.Org Foundation"
[  4370.979]    compiled for 1.21.1.7, module version = 1.21.1
[  4370.979]    Module class: X.Org Video Driver
[  4370.979]    ABI class: X.Org Video Driver, version 25.2
[  4370.980] (II) UnloadModule: "modesetting"
[  4370.980] (II) Unloading modesetting
[  4370.980] (II) Failed to load module "modesetting" (already loaded, 0)
[  4370.980] (II) LoadModule: "fbdev"
[  4370.980] (WW) Warning, couldn't open module fbdev
[  4370.980] (EE) Failed to load module "fbdev" (module does not exist, 0)
[  4370.980] (II) modesetting: Driver for Modesetting Kernel Drivers: kms
[  4370.980] (WW) Falling back to old probe method for modesetting
[  4370.981] (WW) Falling back to old probe method for modesetting
[  4370.981] (EE) No devices detected.
[  4370.981] (EE) 
Fatal server error:
[  4370.981] (EE) no screens found(EE)
```

I've also attached journal log when starting gdm service
[gdm.journal.log.txt](https://github.com/devmfc/debian-on-amlogic/files/11510540/gdm.journal.log.txt)

Please tell me how can I proceed to get desktop working and if any other info or logs are needed.
 Thanks a lot

**Comment by devmfc** [comment]
For S905X4 (SC2) and S905W2 (S4) there is no working mainline HDMI driver at this moment. So boxes based on these SOCs will only work as a headless server. There is a text console however via simpleframebuffer (uses mem area and HDMI set up by bootloader). But for desktop usage: a no-go for now. 

Sorry.

**Reply by ductm208** [reply]
I wish I went to this topic before. I have Tanix W2 S905W2, desktop setup brought me a blinking cursor. Thanks for the information. Btw I am using it as smart home controllers, headless is just fine.

**Comment by evandrojr** [comment]
HDMI on S905X4 is now working. Except for Xorg, that I didn´t manage to make work.

**Reply by Dracoit** [reply]
Hi, is HDMI working on S905X4? in my case only the console is visible.

**Comment by hirotokun4ik** [comment]
After installing desktop-minimal-install.sh it started but the cursor is not visible what could this be connected with (device X96 s905w)

**Comment by easytol0ve** [comment]
you can install desktop on tanix W2 and similar devices

apt install xserver-xorg
apt install xserver-xorg-video-all
apt install lightdm
apt install xinit 

nano /etc/X11/xorg.conf

remove all in xorg.conf and put this

Section "Device"
    #Identifier  "DRM Graphics Acclerated"
    #Driver      "modesetting"
    Identifier "Card0"
    Driver      "fbdev"
    Option      "AccelMethod"    "glamor"     ### "glamor" to enable 3D
    Option      "fbdev"          "/dev/fb0"
    #Option      "DRI"            "2"
    #Option      "Dri2Vsync"      "true"
    #Option      "TripleBuffer"   "true"
EndSection

Section "OutputClass"
    Identifier          "Amlogic"
    MatchDriver         "meson"
    #Driver              "modesetting"
    Option              "PrimaryGPU"  "true"
EndSection

Section "Screen"
	    Identifier "Default Screen"
	    Device "Meson"
	    #Monitor "HDMI"
	    SubSection "Display"
	         Depth 24
	         #Modes "1920x1080" "1440x900" "1280x720" "1280x1024" "1280x960" "1024x768" "800x600" "640x480" "720x400"
       EndSubSection
EndSection

Save

istall your favorite Desktop Environment
 i think lxde more faster than others
apt install lxde

and run startx

**Reply by bemmivend325** [reply]
I tested this now. Use Driver "modesetting" not "fbdev" to get hardware acceleration.
xorg glmark2 387 points
wayland glmark2 600 points

**Reply by petesimon** [reply]
thank you @easytol0ve and @bemmivend325 . i installed ubuntu plucky [v6.12.30](https://github.com/devmfc/debian-on-amlogic/releases/tag/v6.12.30) and lxde desktop on a h96 max W2 (Amloigic S905W2) 2/16. Installed [Brave browser](https://brave.com/linux/).

to allow 1280x720/60 resolution on my hdmi tv, i installed the cvt command
 apt install -y xcvt

i ran xrandr --listmonitors to know the designation of my monitor. i have HDMI-1
you may have another designation such as DP-1 for the line Identifier

and ran the command in a terminal window and a Modeline will show
cvt 1280 720

at the bottom of xorg.conf file, add
Section "Monitor"

add Identifier "HDMI-1"
add the entire line of text Modeline from cvt under Identifier and above Option
add Option "PreferredMode" "1280x720_60"
add EndSection
save file

Logout of graphical desktop, and login again. Open Monitor Settings and
select the new resolution 1280x720_60 then apply and save changes.

here is my xorg.conf

Section "Device"
 Driver "modesetting" ### yes, this works and is faster than "fbdev"
 Identifier "Card0"
 Option "AccelMethod" "glamor" ### "glamor" to enable 3D
 Option "fbdev" "/dev/fb0"
EndSection

Section "OutputClass"
 Identifier "Amlogic"
 MatchDriver "meson"
 Option "PrimaryGPU" "true"
EndSection

Section "Screen"
 Identifier "Default Screen"
 Device "Meson"
 SubSection "Display"
  Depth 24
  Modes "1280x720"
 EndSubSection
EndSection

Section "Monitor"
 Identifier "HDMI-1"
  Modeline "1280x720_60"   74.50  1280 1344 1472 1664  720 723 728 748 -hsync +vsync
   Option "PreferredMode" "1280x720_60"
EndSection

![devmfc-lxde-graphical-brave-browser-720p-IMG_20250629_160010](https://github.com/user-attachments/assets/3916a4dd-ec5d-4432-9639-f47557dc7684)
![devmfc-lxde-graphical-S905W2](https://github.com/user-attachments/assets/ce147ad5-1660-4a24-98d6-a6d5258d5008)

**Comment by bemmivend325** [comment]
It's fast enough for desktop use now on s905x4 meson sc2. I used Ubuntu 24.10 6.12.18 image. Installed labwc, it's GPU accelerated. Using Chromium browser to get hardware acceleration. Firefox was very slow because they want OpenGL 3.2. Panfrost driver works well. I have not tried to get hardware video acceleration working yet.
OpenMW native is playable. I have yet to try box86.
To get sound I used XSDL app on my phone.
Great job Devmfc

**Reply by jojkol** [reply]
> 现在在 s905x4 meson sc2 上，它的速度已经足够快，可以用于桌面端了。我使用的是 Ubuntu 24.10 6.12.18 镜像。安装了 labwc，它支持 GPU 加速。我使用 Chromium 浏览器来获得硬件加速。Firefox 非常慢，因为他们需要 OpenGL 3.2。Panfrost 驱动程序运行良好。我还没有尝试过让硬件视频加速工作。OpenMW 原生驱动可以播放。我还没有尝试过 box86。 为了获得声音，我在手机上使用了 XSDL 应用。Devmfc 做得好！

How did you install the desktop on the s905x4? I have a machine like yours and I failed to install the desktop.

**Reply by bemmivend325** [reply]
I installed labwc window manager, sfwbar panel, pcmanfm file manager, chromium browser, alacritty terminal, not a full desktop environment
launching labwc works by typing labwc into the tty after logging in
for an X11 desktop environment you should follow these instructions https://github.com/devmfc/debian-on-amlogic/discussions/2#discussioncomment-11501760

**Reply by jojkol** [reply]
I followed your method and successfully started the LXDE desktop, but failed when I tried to install the Chromium browser. How did you successfully start the Chromium browser?

**Reply by bemmivend325** [reply]
I removed snap and blocked it. 
I used this third party repository https://launchpad.net/~xtradeb/+archive/ubuntu/apps
